src/main/java/com/contentful/java/cma/model/CMAConstraint.java

```diff
@@ -0,0 +1,245 @@
+package com.contentful.java.cma.model;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+/**
+ * What constraints are applied to this rule?
+ */
+public class CMAConstraint {
+
+  /**
+   * Define a path to a field.
+   */
+  public static class Path {
+    private String doc;
+
+    /**
+     * Set the path.
+     *
+     * @param doc the path to the document, like "fields.name.nopenotok".
+     **/
+    public Path setDoc(String doc) {
+      this.doc = doc;
+      return this;
+    }
+
+    /**
+     * @return the doc part of the path.
+     */
+    public String getDoc() {
+      return doc;
+    }
+
+    /**
+     * @return a human readable string, representing the object.
+     */
+    @Override public String toString() {
+      return "Path { "
+          + "doc = " + getDoc() + " "
+          + "}";
+    }
+  }
+
+  /**
+   * Create an equals object. Used for deciding whether the path to a field containts a given value.
+   */
+  public static class Equals extends ArrayList<Object> {
+    /**
+     * @return which path this equals is targeting.
+     */
+    public Path getPath() {
+      if (size() == 2) {
+        final Object path = this.get(0);
+        if (path instanceof Path) {
+          return (Path) path;
+        } else if (path instanceof Map) {
+          Map mappedPath = (Map) path;
+          final Path objectivicedPath = new Path().setDoc((String) mappedPath.get("doc"));
+          setPath(objectivicedPath);
+          return objectivicedPath;
+        }
+        // missing else for if type is different.
```

Comment:
I'm going to reword this comment to say why a different type is not expected here.